### PR TITLE
Convert RPI to Serial Number

### PR DESCRIPTION
**Pull request by SebastianMaj**
**Why**
Need to convert RPI to Serial Number, read in RockPi serial number, and distinguish between RockPi and Raspberry Pi

**How**
- Replace any references of RPI to serial_number
- Fetch RockPi serial number from file (/proc/device-tree/serial-number) 
- @ilyastrodubtsev helped me discover this file and after looking deeper into it together we realized that this file also exists on the Raspberry Pi and contains just the serial number which means we can avoid the parsing we do with /proc/cpuinfo
- This removes the need to distinguish between RockPi and RaspberryPi as they both have this file in common

**References**
Issue #206

### CODE REVIEW

File: hw_diag/diagnostics/serial_number_diagnostic.py
Comment (marvinmarnold) [inline hw_diag/diagnostics/serial_number_diagnostic.py:9]: This class name should be updated too

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

File: bigquery/bq_schema.json
Comment (marvinmarnold) [inline bigquery/bq_schema.json:54]: Also need to update `MH` to be nullable, right?

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
After this gets approved, please don't merge immediately. This will need to be coordinated with several other PRs.

File: bigquery/bq_schema.json
Comment (SebastianMaj) [reply bigquery/bq_schema.json:54]: Yep I made MH, BCH, BSP all nullable, those are the main values that display numbers and hold weird numbers in the case of an error

Comment (SebastianMaj) [reply bigquery/bq_schema.json:54]: Also no worries, I won't merge until given the ok to merge. Thank you for letting me know

Comment (marvinmarnold) [reply bigquery/bq_schema.json:54]: > Yep I made MH, BCH, BSP all nullable, those are the main values that display numbers and hold weird numbers in the case of an error

My bad, didn't realize these were already nullable. Thanks.

**Review by marvinmarnold — DISMISSED** [review]

**Comment by marvinmarnold** [comment]
@SebastianMaj to get this unblocked by https://github.com/NebraLtd/hm-diag/issues/220, can you update the google cloud [upload logic](https://github.com/NebraLtd/hm-diag/blob/b1210ec87ab347d2a1ae8fe358b30fbf33bcb317/hw_diag/utilities/gcs_shipper.py#L39) so that it still uses `raspberry_pi_serial_number` for the upload?

**Review by marvinmarnold — APPROVED** [review]

**Comment by shawaj** [comment]
> After this gets approved, please don't merge immediately. This will need to be coordinated with several other PRs.

Oops missed this comment sorry

**Comment by vpetersson** [comment]
@shawaj Please don't merge other people's PRs going forward. We have a process for when and how we merge.

**Pull request merged**
